src/mbgl/renderer/sources/render_image_source.cpp

```diff
@@ -133,8 +133,12 @@ void RenderImageSource::update(Immutable<style::Source::Impl> baseImpl_,
     auto dMax = std::max(dx, dy);
     double zoom = std::max(0.0, std::floor(-util::log2(dMax)));
 
-    // Don't bother drawing the ImageSource unless it occupies >4 screen pixels
-    enabled = dMax * std::pow(2.0, transformState.getZoom()) > 2.0 / util::tileSize;
+    // Only enable if the long side of the image is > 2 pixels. Resulting in a
+    // display of at least 2 x 1 px image
+    // dMax is in tile coordinate units at z0, so scale by util::EXTENT and then
+    // by 2^z to get geometry coordinates at the current zoom.
+    // 1 gc unit  = tileSize / extent pixels.
```

Comment:
OK, I think I understand how I got confused now.

To me `GeometryCoordinate` just means a coordinate represented as a pair of `int16_t`s. I didn't pick up that by "GeometryCoordinate" you meant what I'm used to thinking of as "tile unit coordinates" (ie 0 to EXTENT). I also didn't realize that the `TileCoordinate` units were the based on the width of a single tile -- I thought they were EXTENT based. How about something like:

    // Only enable if the long side of the image is > 2 pixels, resulting in a
    // display of at least 2 x 1 px image
    // dMax is in TileCoordinate units at z0, so one unit represents the width of one z0 tile
    // To convert to pixels, multiply by util::tileSize and then scale by 2^z to match current zoom